oasislmf/preparation/il_inputs.py

```diff
@@ -563,9 +563,22 @@ def __merge_gul_and_account(gul_inputs_df, accounts_df, fm_terms, oed_hierarchy)
     cond_class = oed_hierarchy['condclass']['ProfileElementName'].lower()
     loc_num = oed_hierarchy['locnum']['ProfileElementName'].lower()
 
-    policy_df = accounts_df.drop_duplicates(subset=[portfolio_num, acc_num, policy_num, 'layer_id']).drop(columns=cond_tag)
-    cond_df = accounts_df[[portfolio_num, acc_num, cond_tag]].drop_duplicates()
-    all_cond_policy =  pd.merge(policy_df, cond_df, on=[portfolio_num, acc_num])
+    if cond_class in accounts_df.columns:
+        accounts_df['polcondclass'] = (accounts_df.groupby([portfolio_num, acc_num, policy_num, 'layer_id'], sort=False)
+                                                  [cond_class]
+                                                  .transform(max))
+    else:
+        accounts_df[['polcondclass', cond_class]] = 0
+
+    if 'condpriority' not in accounts_df.columns:
+        accounts_df['condpriority'] = 0
+    accounts_df['condpriority'].fillna(0, inplace=True)
+
+    policy_df = accounts_df.drop_duplicates(subset=[portfolio_num, acc_num, policy_num, 'layer_id']).drop(columns=[cond_tag, 'condpriority'])
```

Comment:
please add a short comment before these manipulations of `accounts_df` to explain what we need to achieve.